### PR TITLE
[TW-402] Update comments for ForageAPiError

### PR DESCRIPTION
**Pull request by kimberleehowley**
## What

This PR adds comments to `ForageAPIResponse.kt`, with the goal of improving the documentation for Forage Errors. 

@dleis612 , hi! @djoksimo and I discussed this one IRL 🥳 , and we thought that `ForageApiError` and `ForageErrorObj` should be `internal`, so this PR adds that too. Please let me know of any objections? 

## Why

Better docs -> More merchants onboard faster. 

## Test Plan

- ✅ | The reviewer should manually test the changes in this PR. 
Check out my branch and run: 

```
./gradlew dokkaHtml
npx http-serve reference-docs
```

### CODE REVIEW

File: forage-android/src/main/java/com/joinforage/forage/android/network/model/ForageApiResponse.kt [hunk 1]
```diff
@@ -6,9 +6,23 @@ internal val UnknownErrorApiResponse = ForageApiResponse.Failure.fromError(
     ForageError(500, "unknown_server_error", "Unknown Server Error")
 )
 
+/**
+ * A model that represents the possible types of responses from the Forage API.
+ *
```
Comment (djoksimo) [inline forage-android/src/main/java/com/joinforage/forage/android/network/model/ForageApiResponse.kt:11]: Can remove extra `*` newline on line 11

File: forage-android/src/main/java/com/joinforage/forage/android/network/model/ForageApiResponse.kt [hunk 2]
```diff
@@ -6,9 +6,23 @@ internal val UnknownErrorApiResponse = ForageApiResponse.Failure.fromError(
...
+ */
 sealed class ForageApiResponse<out T> {
+    /**
+     * A model that represents a success response from the API.
+     *
```
Comment (djoksimo) [inline forage-android/src/main/java/com/joinforage/forage/android/network/model/ForageApiResponse.kt:16]: can remove newline

Comment (djoksimo) [reply forage-android/src/main/java/com/joinforage/forage/android/network/model/ForageApiResponse.kt:16]: Can we add some additional context, like "In most cases `data` is a string representation of a JSON object from the Forage API"

 and consider quoting an example (either when we introduce `@sample` or by simply mentioning:

`if (response is ForageApiResponse.Success) { response.data // { "ref": "abcde123", ... }`

File: forage-android/src/main/java/com/joinforage/forage/android/network/model/ForageApiResponse.kt [hunk 3]
```diff
@@ -6,9 +6,23 @@ internal val UnknownErrorApiResponse = ForageApiResponse.Failure.fromError(
...
+     */
     data class Success<out T>(val data: T) : ForageApiResponse<T>()
 
+    /**
+     * A model that represents a failure response from the API.
+     *
+     * @property errors A list of [ForageError] instances that you can unpack to programmatically
+     * handle the error and display the appropriate customer-facing message.
+     */
     data class Failure(val errors: List<ForageError>) : ForageApiResponse<Nothing>() {
```
Comment (djoksimo) [inline forage-android/src/main/java/com/joinforage/forage/android/network/model/ForageApiResponse.kt:26]: Can we link to relevant Forage docs that talk about displaying customer-facing error copy?

**Review by djoksimo — CHANGES_REQUESTED** [review]

File: forage-android/src/main/java/com/joinforage/forage/android/network/model/ForageApiResponse.kt (hunk 1 above)
Comment (kimberleehowley) [reply forage-android/src/main/java/com/joinforage/forage/android/network/model/ForageApiResponse.kt:11]: Yes of course! I had hoped the spotlessApply would've caught these things 🤔

File: forage-android/src/main/java/com/joinforage/forage/android/network/model/ForageApiResponse.kt (hunk 3 above)
Comment (kimberleehowley) [reply forage-android/src/main/java/com/joinforage/forage/android/network/model/ForageApiResponse.kt:26]: Hmmm this is one I want to think about! That particular doc is specific to BRDs, so I am worried it's a little confusing. I can link for now, but need to think on this more longterm as well.

**Review by djoksimo — APPROVED** [review]
Would be good for @devinmorgan or @dleis612 to confirm whether it's safe to make `ForageErrorObj` internal

**Comment by kimberleehowley** [comment]
> Would be good for @devinmorgan or @dleis612 to confirm whether it's safe to make `ForageErrorObj` internal

Thank you, @djoksimo ! I confirmed with @devinmorgan IRL that this is good to be internal. I'll merge the PR.

**Pull request merged**